beatport_importer.user.js

```diff
@@ -24,12 +24,23 @@ if (!unsafeWindow) unsafeWindow = window;
 $(document).ready(function () {
     MBImportStyle();
 
-    let release_url = window.location.href.replace('/?.*$/', '').replace(/#.*$/, '');
-    let release = retrieveReleaseInfo(release_url);
-    insertLink(release, release_url);
+    $.getJSON(`https://www.beatport.com/api/v4/catalog/releases/${ProductDetail.id}`).done(release_data => {
+        let release_url = window.location.href.replace('/?.*$/', '').replace(/#.*$/, '');
+        let release = retrieveReleaseInfo(release_url, release_data);
+
+        const track_urls = release_data.tracks.map(url => url.replace('api.beatport.com', 'www.beatport.com/api')).reverse();
```

Comment:
1. Since you are already rewriting the API URL, you could also drop the trailing slash from the URL.
   Currently I'm seeing redirects happening for every track API call which can be avoided:
   https://www.beatport.com/api/v4/catalog/tracks/17674616/ [HTTP/2 301 Moved Permanently] -> https://www.beatport.com/api/v4/catalog/tracks/17674616 [HTTP/2 200 OK]
2. Reversing the order of the track API URLs in order to get the correct track order seems a bit unreliable to me, I would suggest to make use of the `number` attribute in each track API response instead (under the assumption that all Beatport releases only have a single medium). See my comments below.